02-household-queries/optimize_encoder.py

```diff
@@ -0,0 +1,162 @@
+import os
+from bs4 import BeautifulSoup
+import dotenv
+from langchain.docstore.document import Document
+from langchain_text_splitters import RecursiveCharacterTextSplitter #, NLTKTextSplitter, SpacyTextSplitter
+import json
+from langchain_community.embeddings import (
+    SentenceTransformerEmbeddings,
+    HuggingFaceEmbeddings,
+)
+import chromadb
+from chromadb.config import Settings
+
+# from langchain_google_genai import ChatGoogleGenerativeAI, GoogleGenerativeAIEmbeddings
+
+from llm import ollama_client
+from retrieval import create_retriever
+from langchain_community.vectorstores import Chroma
+# import nltk
+# import spacy
+
+dotenv.load_dotenv()
+
+_llm_model_name = os.environ.get("LLM_MODEL_NAME", "mistral")
+
+
+llm = {
+    "gpt4all_path": "./mistral-7b-instruct-v0.1.Q4_0.gguf",
+    "ollama": ollama_client(_llm_model_name, settings={"temperature":0.1})
+}
```

Comment:
Remove since LLMs aren't being used for testing the retrieval component.
Also don't do optional initialization outside of a function because `ollama_client` gets automatically called and downloads the model unnecessarily:
```python
    "ollama": ollama_client(_llm_model_name, settings={"temperature":0.1})
```